sdk/servicebus/azure-servicebus/samples/sync_samples/client_identity_authentication.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+"""
+An example to show authentication using credentials defined by azure.identity library.
+
+EnvironmentCredential is capable of authenticating as a service principal using a client secret or a certificate, or as
+a user with a username and password. Configuration is attempted in this order, using these environment variables:
+
+Service principal with secret:
+  - **AZURE_TENANT_ID**: ID of the service principal's tenant. Also called its 'directory' ID.
+  - **AZURE_CLIENT_ID**: the service principal's client ID
+  - **AZURE_CLIENT_SECRET**: one of the service principal's client secrets
+
+Service principal with certificate:
+  - **AZURE_TENANT_ID**: ID of the service principal's tenant. Also called its 'directory' ID.
+  - **AZURE_CLIENT_ID**: the service principal's client ID
+  - **AZURE_CLIENT_CERTIFICATE_PATH**: path to a PEM-encoded certificate file including the private key. The
+    certificate must not be password-protected.
+
+User with username and password:
+  - **AZURE_CLIENT_ID**: the application's client ID
+  - **AZURE_USERNAME**: a username (usually an email address)
+  - **AZURE_PASSWORD**: that user's password
+  - **AZURE_TENANT_ID**: (optional) ID of the service principal's tenant. Also called its 'directory' ID.
+    If not provided, defaults to the 'organizations' tenant, which supports only Azure Active Directory work or
+    school accounts.
+
+Please refer to azure.identity library for detailed information.
+
+This sample also shows the process of utilizing a different credential object, in this case, DefaultAzureCredential,
+both to demonstrate the ease of adjusting authentication, and to surface another method for doing so.
+"""
+
+import os
+from azure.servicebus import ServiceBusClient, Message
+from azure.identity import EnvironmentCredential
+
+
+FULLY_QUALIFIED_NAMESPACE = os.environ['SERVICE_BUS_NAMESPACE']
+QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+credential = EnvironmentCredential()
+
+# Note: One has other options to specify the credential.  For instance, DefaultAzureCredential.
```

Comment:
Did you want to put this disclaimer in async sample too?

Reply:
added in async sample